newIDE/app/public/External/Utils/pathEditor.js

```diff
@@ -0,0 +1,132 @@
+const electron = require('electron');
+const fs = require('fs');
+const remote = electron.remote;
+const {
+    dialog
+} = remote;
+
+const headerStyle = {
+    saveFolderLabel: 'height:27px;color:SlateGrey;float: left;margin-left: 2px;margin-top: 10px; font-size:15px;',
+    nameInput: 'font-family:"Courier New";height:27px;width:90px;color:SlateGrey;float:left;margin-left: 2px;padding:4px;margin-top: 4px;font-size:15px;border: 2px solid #e5cd50;border-radius: 3px;  ',
+    fileExistsLabel: 'height:27px;color:blue;float: left;margin-left: 2px;margin-top: 10px; font-size:15px;',
+
+    saveButton: 'height:27px;float:right;margin-left:2px;margin-right:4px;border: 2px solid DeepSkyBlue;border-radius: 1px;margin-top: 5px;background-color:white;',
+    cancelButton: 'height:27px;float:right;margin-right:2px;border: 2px solid DeepSkyBlue;border-radius: 1px;margin-top: 5px;background-color:white;',
+    setFolderButton: 'height:27px;float:right;margin-left:2px;margin-right:4px;border: 2px solid DeepSkyBlue;border-radius: 1px;margin-top: 5px;background-color:white;',
+};
+
+let saveFolderLabel,
+    setFolderButton,
+    nameInput,
+    saveButton,
+    cancelButton,
+    fileExistsLabel = null
+
+export let saveOptions = {}
```

Comment:
I can make updateBasePath return save options? It will prevent a lot of variable passing around.
The only place where its needed is in the save function of the editor

Reply:
Mmmm, are you sure updateBasePath should return it? What is the responsibility of the function? According to the name, it should update the UI of the base path, no more no less.

You might want to split it into two functions:
* one that return saveOptions.
* one that take into parameters some DOM elements and update them (nameInput, saveFolderLabel, etc...)

You could almost name the second function "renderXXX" (a bit like in React ;)). "updateXXX" is fine too.